File: s11_Retrofit/KriptoPara/app/src/main/java/com/omeryildizce/kriptopara/service/CryptoAPI.java
package com.omeryildizce.kriptopara.service;

import com.omeryildizce.kriptopara.model.CryptoModel;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.Call;
import retrofit2.http.GET;

public interface CryptoAPI {
    @GET("atilsamancioglu/K21-JSONDataSet/master/crypto.json")
    Observable<List<CryptoModel>> getData();
    //Call<List<CryptoModel>> getData();
}

File: s25_Coroutine/KriptoPara/app/src/main/java/com/omeryildizce/kriptopara/service/CryptoApi.kt
package com.omeryildizce.kriptopara.service

import com.omeryildizce.kriptopara.model.CryptoModel
import io.reactivex.Observable
import io.reactivex.Observer
import retrofit2.Call
import retrofit2.Response
import retrofit2.http.GET

interface CryptoApi {
    // GET, POST, UPDATE, DELETE
    @GET("atilsamancioglu/K21-JSONDataSet/master/crypto.json")
    //fun getData() : Observable<List<CryptoModel>>
    // fun getData(): Call<List<CryptoModel>>
     suspend fun getData(): Response<List<CryptoModel>>

}